pkg/apply/prune/prune.go

```diff
@@ -157,6 +157,11 @@ func preventDeleteAnnotation(annotations map[string]string) bool {
 				return true
 			}
 		}
+		if annotation == common.LifecycleDeleteAnnotation {
```

Comment:
It seems like we could create a data structure which would contain the different combinations or annotation and value we accept to prevent deletion instead of duplicating the nested if statements here?

Reply:
Moved this check into a common function. PTAL.